timemachine/fe/free_energy.py

```diff
@@ -482,20 +487,17 @@ def run_sequential_sims_given_initial_states(
             stored_frames.append(cur_frames)
             stored_boxes.append(cur_boxes)
 
-        # construct EnergyDecomposedState for current lamb_idx,
-        # but keep no more than 2 of these states in memory at once
-        if lamb_idx >= 2:
-            tmp_states[lamb_idx - 2] = None
-
         bound_impls = [p.bound_impl(np.float32) for p in initial_state.potentials]
         cur_batch_U_fns = get_batch_u_fns(bound_impls, temperature)
 
-        tmp_states[lamb_idx] = EnergyDecomposedState(cur_frames, cur_boxes, cur_batch_U_fns)
+        state = EnergyDecomposedState(cur_frames, cur_boxes, cur_batch_U_fns)
 
         # analysis that depends on current and previous state
-        if lamb_idx > 0:
-            state_pair = [tmp_states[lamb_idx - 1], tmp_states[lamb_idx]]
+        if prev_state:
+            state_pair = [prev_state, state]
```

Comment:
Motivation for the changes here was to prevent a typechecking error that appears upon annotating this function (`state_pair` has type `Tuple[Optional[EnergyDecomposedState], Optional[EnergyDecomposedState]]`, which is incompatible with the input of `compute_energy_decomposed_u_kln`)